scripts/Connector/Connector_JST/conn_jst_vh_tht_top.py

```diff
@@ -149,11 +156,28 @@ def generate_one_footprint(pins, series_params, configuration):
     else:
         optional_pad_params['tht_pad1_shape'] = Pad.SHAPE_ROUNDRECT
 
-    kicad_mod.append(PadArray(
-        pincount=pins, x_spacing=pitch,
-        type=Pad.TYPE_THT, shape=shape,
-        size=pad_size, drill=drill, layers=Pad.LAYERS_THT,
-        **optional_pad_params))
+    if post_omitted:
```

Comment:
Why not use the exclude pin parameter of the pad array?

https://github.com/pointhi/kicad-footprint-generator/blob/f84ae65fdbc720440412c36a69a591330ae5bc50/KicadModTree/nodes/specialized/PadArray.py#L107